pkg/apis/kubernetes/v1/example_test.go

```diff
@@ -0,0 +1,37 @@
+package v1
+
+import (
+	"context"
+	"log"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/client"
+	"go.anx.io/go-anxcloud/pkg/utils/pointer"
+)
+
+func Example() {
+	a, err := api.NewAPI(api.WithClientOptions(client.AuthFromEnv(false)))
+	if err != nil {
+		log.Fatalf("failed to initialize API client: %s", err)
+	}
+
+	cluster := Cluster{Name: "example", NeedsServiceVMs: pointer.Bool(true)}
+
+	if err := a.Create(context.TODO(), &cluster); err != nil {
+		log.Fatalf("failed to create cluster: %s", err)
+	}
+
+	if err := cluster.AwaitCompletion(context.TODO(), a); err != nil {
+		log.Fatalf("failed to await cluster creation: %s", err)
+	}
+
+	nodePool := NodePool{Name: "example-np-00", Cluster: cluster, Replicas: 1, Memory: 2147483648, DiskSize: 10737418240}
```

Comment:
I suggest using powers of 2 for those memory and disk size numbers (Gi = 2**30).

Reply:
this is still to do @marioreggiori

Reply:
@LittleFox94 @toothstone wasn't aware that Golang does not have the ** (pow) expression and decided against an additional import (math pkg).